src/sentry/sentry_metrics/consumers/indexer/batch.py

```diff
@@ -91,14 +100,13 @@ def __init__(
     @metrics.wraps("process_messages.extract_messages")
     def _extract_messages(self) -> None:
         self.skipped_offsets: Set[PartitionIdxOffset] = set()
-        self.parsed_payloads_by_offset: MutableMapping[PartitionIdxOffset, IngestMetric] = {}
+        self.parsed_payloads_by_offset: MutableMapping[PartitionIdxOffset, ParsedMessage] = {}
```

Comment:
Just want to make sure that we're okay with removing `IngestMetric` here completely. Maybe @lynnagara would have thoughts?

In case this is an issue, an alternative would be to just have the `ParsedMessage` type be intermediate. So, we would initially have the `IngestMetric` schema, and then we would apply transformations to the payload so that it becomes the `ParsedMessage` payload type.

Reply:
I think with the alternative you suggested `self.parsed_payloads_by_offse`t would still need to be of type `MutableMapping[PartitionIdxOffset, ParsedMessage]` because it **is** the alias for our intermediate representation.
We can set the `parsed_payload` (line 109) to the `IngestMetric` type, though, and then once we do (line 130)
```python
parsed_payload["use_case_id"] = use_case_id
```
We can cast it to the `parsed_payload` type from the `IngestMetric` type somehow

Reply:
@lynnagara confirming that removing `IngestMetric` as above is fine?

Reply:
Looking at the ParsedMessage definition, isn't it saying that the topic consumed by the indexer that is being decoded here `ingest-metrics` has field `use_case_id` which is a required string? But we know that Relay actually doesn't produce messages with this property, how can that be true?

Reply:
Same question for line 187 `use_case_id = message["use_case_id"]` - how does it work if there's no such property there?

Reply:
@lynnagara So I think the intention behind this change was: we take the payload we get from Relay, which doesn't have `use_case_id` in it, correct. But this payload implicitly has the use_case_id in the `name` field of the payload (Relay formats it as a MRI in the name field). What we're doing is extracting that `use_case_id` and adding it explicitly to the payload. We're doing this essentially in the constructor upon receipt of a Batch of messages. 

So Relay is sending something which is almost always a valid MRI with a valid use case (Relay does filtering, and the generic metrics interface I'm working on would block any invalid use case). In the very off chance that the MRI or use case in the MRI is invalid, we can throw an error and stop processing, or send the message to DLQ whenever that's available. 

If the approach with using the `ParsedMessage` type directly I described in the first paragraph here doesn't make sense to you, how about if we used `ParsedMessage` as only an intermediate type? i.e. the payload gets initialized as a `IngestMetric` message and then gets the `use_case_id` field and transforms into a `ParsedMessage` type?